.travis/script_macos.sh

```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+
+if [[ "$PYTHON_INTERP" == "python2.7" ]]; then
+  export PY2_CMAKE_ARGS="-DPYTHON_EXECUTABLE=/usr/local/bin/python -DPYTHON_LIBRARY=/usr/local/opt/python@2/Frameworks/Python.framework/Versions/2.7/lib/libpython2.7.dylib"
```

Comment:
I'm not a big fan of this; if there is something wrong in the Python detection in CMake, we should fix it in the CMake scripts, unless this is a type of misconfiguration that only happens on Travis and nowhere else.

Reply:
When searching for Python 2, CMake seemingly refuses to select any install other than the system python. This seems to be a long-standing issue and I haven't been able to find a clean solution. Maybe @CFSworks can chime in on this.